### PR TITLE
[JSDK-72] Fixes spotless on recent java version and enable CI matrix on build jobs

### PR DESCRIPTION
**Pull request by dili91**
# Description

- Adds gradle options to have spotless happy on JDK 16+. 
- Adds a matrix of JDK versions 8,11,15,17 when building the project, and updates the main one to 17
- A few doc fixes

This change won't trigger any release 

## Type of change

Please select multiple options if required.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where necessary
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the relevant documentation

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,26 +9,36 @@ on:
...
         uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      - name: Fix Gradle options on Java < 9
+        if: ${{ matrix.java < 9 }}
+        run: |
+          # removes jvm args
+          sed -ie s/org\.gradle\.jvmargs/_ignored/ gradle.properties
       - name: Get project version
         id: get_project_version
         run: |
-          PROJECT_VERSION=$(gradle properties -q | grep "version:" | awk '{print $2}')
```
Comment (dili91) [inline .github/workflows/build.yml:29]: this was an oversight

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Acceptance tests
```
Comment (dili91) [inline .github/workflows/coverage.yml:1]: was an oversight

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -19,13 +19,13 @@ jobs:
...
         uses: actions/setup-java@v2
         with:
-          java-version: '8'
+          java-version: '17'
           distribution: 'adopt'
           cache: 'gradle'
       - name: Validate Gradle wrapper
         uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
       - name: Test coverage
-        run: ./gradlew jacocoTestReport coveralls
```
Comment (dili91) [inline .github/workflows/coverage.yml:28]: was an oversight

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -9,26 +9,36 @@ on:
...
+      - name: Set up JDK ${{ matrix.java }}
         uses: actions/setup-java@v2
         with:
-          java-version: '8'
+          java-version: ${{ matrix.java }}
           distribution: 'adopt'
           cache: 'gradle'
       - name: Validate Gradle wrapper
         uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      - name: Fix Gradle options on Java < 9
```
Comment (dili91) [inline .github/workflows/build.yml:31]: the additional gradle options that I had to add to have the linter working are not compatible with versions of Java below 9 😠

File: .github/workflows/acceptance-tests.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: Set up JDK 8
         uses: actions/setup-java@v2
         with:
-          java-version: '8'
+          java-version: '17'
```
Comment (nico-incubiq) [inline .github/workflows/acceptance-tests.yml:23]: Update the comment line 20 if we're running the acceptance-tests against java 17

File: .github/workflows/release-final.yml [hunk 6]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: Set up JDK 8
         uses: actions/setup-java@v2
         with:
-          java-version: '8'
+          java-version: '17'
```
Comment (nico-incubiq) [inline .github/workflows/release-final.yml:39]: and here, update line 36

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -9,26 +9,36 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
     name: Build and tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        java: [ 8, 11, 15, 17 ]
```
Comment (nico-incubiq) [inline .github/workflows/build.yml:17]: why 15 but not 18? would be useful to test against the latest version out there in additions to the LTSs.

**Pull request merged**